src/master/master.ts

```diff
@@ -459,7 +459,30 @@ export class Master {
     }
   }
 
-  async onChatMessage(matchID, chatMessage) {
+  async onChatMessage(
+    matchID: string,
+    chatMessage: ChatMessage,
+    credentials: string | undefined
+  ): Promise<void | { error: string }> {
+    const key = matchID;
+    let metadata: Server.MatchData | undefined;
+
+    if (StorageAPI.isSynchronous(this.storageAPI)) {
+      ({ metadata } = this.storageAPI.fetch(key, { metadata: true }));
+    } else {
+      ({ metadata } = await this.storageAPI.fetch(key, { metadata: true }));
+    }
+    if (this.auth && chatMessage.sender !== undefined) {
```

Comment:
@TomasVidhall I’m not 100% sure about allowing spectators to post messages without authentication, although this is an improvement over allowing them to claim to be a player. Is this a functionality you need? In general I would lean towards only permitting players to post, but am open to being more flexible if you definitely need it (or perhaps adding a server option to enable spectator chat).